geo/src/algorithm/triangulate_delaunay.rs

```diff
@@ -0,0 +1,645 @@
+use std::{fmt, fmt::Debug};
+
+use crate::coord;
+use crate::{BoundingRect, Coord, GeoFloat, Line, MultiPoint, Polygon, Triangle};
+
+/// The default expansion value for creating the super triangle.
+/// A bounding rectangle is computed for the points and the super triangle
+/// is expanded by this value by default.
+pub const DEFAULT_SUPER_TRIANGLE_EXPANSION: f64 = 20.;
+
+type Result<T> = std::result::Result<T, DelaunayTriangulationError>;
+
+/// Delaunay Triangulation for a given set of points using the
+/// [Bowyer](https://doi.org/10.1093%2Fcomjnl%2F24.2.162)-[Watson](https://doi.org/10.1093%2Fcomjnl%2F24.2.167)
+/// algorithm
+pub trait TriangulateDelaunay<T: GeoFloat> {
+    /// # Example
+    ///
+    /// ```rust
+    /// use geo::{coord, polygon, Triangle, TriangulateDelaunay};
+    ///
+    /// let points = polygon![
+    ///     (x: 0., y: 0.),
+    ///     (x: 1., y: 2.),
+    ///     (x: 2., y: 4.),
+    ///     (x: 2., y: 0.),
+    ///     (x: 3., y: 2.),
+    ///     (x: 4., y: 0.),
+    /// ];
+    ///
+    /// let tri_force = points.delaunay_triangulation().unwrap();
+    ///
+    /// assert_eq!(vec![
+    ///     Triangle(
+    ///         coord! {x: 1., y: 2.},
+    ///         coord! {x: 0., y: 0.},
+    ///         coord! {x: 2., y: 0.},
+    ///     ),
+    ///     Triangle(
+    ///         coord! {x: 2., y: 4.},
+    ///         coord! {x: 1., y: 2.},
+    ///         coord! {x: 3., y: 2.},
+    ///     ),
+    ///     Triangle(
+    ///         coord! {x: 1., y: 2.},
+    ///         coord! {x: 2., y: 0.},
+    ///         coord! {x: 3., y: 2.},
+    ///     ),
+    ///     Triangle(
+    ///         coord! {x: 3., y: 2.},
+    ///         coord! {x: 2., y: 0.},
+    ///         coord! {x: 4., y: 0.},
+    ///     )],
+    ///     tri_force
+    /// );
+    ///
+    /// ```
+    ///
+    fn delaunay_triangulation(&self) -> Result<Vec<Triangle<T>>>;
+}
+
+impl<T: GeoFloat> TriangulateDelaunay<T> for Polygon<T>
+where
+    f64: From<T>,
+{
+    fn delaunay_triangulation(&self) -> Result<Vec<Triangle<T>>> {
+        let super_triangle = DelaunayTriangle(create_super_triangle(self)?);
+
+        let mut triangles: Vec<DelaunayTriangle<T>> = vec![super_triangle.clone()];
+        let mut processed_points: Vec<Coord<T>> = Vec::new();
+        for pt in self.exterior().coords() {
+            // If we have already added the point, do not process again
+            // to avoid producing intersecting triangles
+            if !processed_points.contains(pt) {
+                add_coordinate(&mut triangles, pt);
+                processed_points.push(*pt);
+            }
+        }
+        let triangles = remove_super_triangle(&triangles, &super_triangle);
```

Comment:
Questions: 

- Can it happen that the `super_triangle` gets removed during the algorithm?
- If not, can it happen that the `super_triangle` gets moved away from the first position to somewhere else in the Vec

Depending on the answers, the removal might be easier than a full search through the whole vec as done in `remove_super_triangle`

Reply:
No the `super_triangle` isn't removed during the process.  The scaffolding provided by the `super_triangle` will remain as well as other triangles formed from it, ![an example](https://en.wikipedia.org/wiki/Bowyer%E2%80%93Watson_algorithm#/media/File:Bowyer-Watson_6.png).  In this diagram the `super_triangle` and corresponding artefacts are shown in red while the final delaunay triangulation is shown in below.

The process of removing the `super_triangle` also requires removing triangles that share vertices with the `super_triangle` and these triangles can be located anywhere in the vec.